docs/hooks.md

```diff
@@ -0,0 +1,242 @@
+
+```
+kind: Plan
+apiVersion: forklift.konveyor.io/v1beta1
+metadata:
+  name: test
+  namespace: konveyor-forklift
+spec:
+  vms:
+    - id: vm-2861
+      hooks:
+        - hook:
+            namespace: konveyor-forklift
+            name: playbook
+          step: PreHook
+...
+```
+
+# Adding a Hook CR
+The Hook CR represents a hook and an example is provided below. The playbook is base64 encoded.
+
+To encode a playbook cat a file and pipe it for base64, for example `cat playbook.yml | base64 -w0`.
+
+It is also possible to use a here doc:
+```
+cat << EOF | base64 -w0
+- hosts: localhost
+  tasks:
+  - debug:
+      msg: test
+EOF
+```
+
+To decode an attached playbook retrieve the resource with custom output and pipe it to base64, for example `oc get -n konveyor-forklift hook playbook -o go-template='{{ .spec.playbook }}' | base64 -d`.
+
+Hook Example:
+```
+apiVersion: forklift.konveyor.io/v1beta1
+kind: Hook
+metadata:
+  name: playbook
+  namespace: konveyor-forklift
+spec:
+  image: quay.io/konveyor/hook-runner
+  playbook: LSBuYW1lOiBNYWluCiAgaG9zdHM6IGxvY2FsaG9zdAogIHRhc2tzOgogIC0gbmFtZTogTG9hZCBQbGFuCiAgICBpbmNsdWRlX3ZhcnM6CiAgICAgIGZpbGU6IHBsYW4ueW1sCiAgICAgIG5hbWU6IHBsYW4KICAtIG5hbWU6IExvYWQgV29ya2xvYWQKICAgIGluY2x1ZGVfdmFyczoKICAgICAgZmlsZTogd29ya2xvYWQueW1sCiAgICAgIG5hbWU6IHdvcmtsb2FkCgojICAtIGFkZF9ob3N0OgojICAgICAgbXNnOiAie3sgd29ya2xvYWQudm0uaXBhZGRyZXNzIH19IgojCiMgIC0gZGVidWc6CiMgICAgICBtc2c6ICJ1c2VyOiB7eyBzc2hfY3JlZGVudGlhbHMucmVzb3VyY2VzWzBdLmRhdGEudXNlciB8IGI2NGRlY29kZSB9fSwgcGFzc3dvcmQ6IHt7IHNzaF9jcmVkZW50aWFscy5yZXNvdXJjZXNbMF0uZGF0YS5wYXNzd29yZCB8IGI2NGRlY29kZSB9fSIKCiAgLSBuYW1lOiBFbnN1cmUgZGlyZWN0b3J5IGV4aXN0cwogICAgZmlsZToKICAgICAgcGF0aDogfi8uc3NoCiAgICAgIHN0YXRlOiBkaXJlY3RvcnkKICAgICAgbW9kZTogMDc1MAoKICAtIGs4c19pbmZvOgogICAgICBhcGlfdmVyc2lvbjogdjEKICAgICAga2luZDogU2VjcmV0CiAgICAgIG5hbWU6IHNzaC1jcmVkZW50aWFscwogICAgICBuYW1lc3BhY2U6IGtvbnZleW9yLWZvcmtsaWZ0CiAgICByZWdpc3Rlcjogc3NoX2NyZWRlbnRpYWxzCgogIC0gbmFtZTogQ3JlYXRlIFNTSCBrZXkKICAgIGNvcHk6CiAgICAgIGRlc3Q6IH4vLnNzaC9pZF9yc2EKICAgICAgY29udGVudDogInt7IHNzaF9jcmVkZW50aWFscy5yZXNvdXJjZXNbMF0uZGF0YS5rZXkgfCBiNjRkZWNvZGUgfX0iCiAgICAgIG1vZGU6IDA2MDAKCiAgLSBhZGRfaG9zdDoKICAgICAgbmFtZTogInt7IHdvcmtsb2FkLnZtLmlwYWRkcmVzcyB9fSIKICAgICAgYW5zaWJsZV91c2VyOiByb290CiAgICAgIGdyb3Vwczogdm1zCgotIGhvc3RzOiB2bXMKICB0YXNrczoKICAtIG5hbWU6IEluc3RhbGwgY2xvdWQtaW5pdAogICAgZG5mOgogICAgICBuYW1lOgogICAgICAtIGNsb3VkLWluaXQKICAgICAgc3RhdGU6IGxhdGVzdAoKICAtIG5hbWU6IENyZWF0ZSBUZXN0IEZpbGUKICAgIGNvcHk6CiAgICAgIGRlc3Q6IC90ZXN0LnR4dAogICAgICBjb250ZW50OiAiSGVsbG8gV29ybGQiCiAgICAgIG1vZGU6IDA2NDQK
+
+```
+
+# Storing additional information in secrets
+If you wish to access additional information stored in secrets from within the hook container you may, but you will need to provide additional access to the default account in the konveyor-forklift namespace. An example role and rolebinding to provide acecss to secrets is included here.
```

Comment:
I will tweak your example to set the Hook SA to the controller SA but we should word this to encourage users to do this instead of granting read on secret to the default SA.

Reply:
I removed the role(binding) stuff, added the SA to the example, and mentioned the possibility of changing it.